the-badass-todo-list-part-1/input.js

```diff
@@ -18,16 +20,20 @@
         };
         // To prevent user saving multiple times.
         input.disabled = true;
+        // Quiz #2
         saveData(data, tidyUp);
       }
     });
   }
 
+  // Quiz #2
   function saveData(todoItem, afterSave) {
     console.log('saved');
+    // Quiz #2
     afterSave();
   }
 
+  // Quiz #2
   function fetchData(afterFetch) {
```

Comment:
You got it. That's why we need **promise** and other much advanced techniques like `async` function to solve the problem. Unfortunately, not every front-end developer like to embrace new techs. You will find people is almost like to **hate** to solve expressiveness problems.